File: dart/lib/api/geocoding_api.dart
part of swagger.api;



class GeocodingApi {
  final ApiClient apiClient;

  GeocodingApi([ApiClient apiClient]) : apiClient = apiClient ?? defaultApiClient;

  /// Execute a Geocoding request
  ///
  /// This endpoint provides forward and reverse geocoding. For more details, review the official documentation at: https://graphhopper.com/api/1/docs/geocoding/ 
  Future<GeocodingResponse> geocodeGet(String key, { String q, String locale, int limit, bool reverse, String point, String provider }) async {
    Object postBody = null;

    // verify required params are set
    if(key == null) {
     throw new ApiException(400, "Missing required param: key");
    }

    // create path and map variables
    String path = "/geocode".replaceAll("{format}","json");

    // query params
    List<QueryParam> queryParams = [];
    Map<String, String> headerParams = {};
    Map<String, String> formParams = {};
    if(q != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "q", q));
    }
    if(locale != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "locale", locale));
    }
    if(limit != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "limit", limit));
    }
    if(reverse != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "reverse", reverse));
    }
    if(point != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "point", point));
    }
    if(provider != null) {
      queryParams.addAll(_convertParametersForCollectionFormat("", "provider", provider));
    }
      queryParams.addAll(_convertParametersForCollectionFormat("", "key", key));
    
    List<String> contentTypes = [];

    String contentType = contentTypes.length > 0 ? contentTypes[0] : "application/json";
    List<String> authNames = [];

    if(contentType.startsWith("multipart/form-data")) {
      bool hasFields = false;
      MultipartRequest mp = new MultipartRequest(null, null);
      
      if(hasFields)
        postBody = mp;
    }
    else {
          }

    var response = await apiClient.invokeAPI(path,
                                             'GET',
                                             queryParams,
                                             postBody,
                                             headerParams,
                                             formParams,
                                             contentType,
                                             authNames);

    if(response.statusCode >= 400) {
      throw new ApiException(response.statusCode, response.body);
    } else if(response.body != null) {
      return apiClient.deserialize(response.body, 'GeocodingResponse') as GeocodingResponse ;
    } else {
      return null;
    }
  }
}

File: ruby-sinatra/api/geocoding_api.rb
require 'json'


MyApp.add_route('GET', '/api/1/geocode', {
  "resourcePath" => "/Geocoding",
  "summary" => "Execute a Geocoding request",
  "nickname" => "geocode_get", 
  "responseClass" => "GeocodingResponse", 
  "endpoint" => "/geocode", 
  "notes" => "This endpoint provides forward and reverse geocoding. For more details, review the official documentation at: https://graphhopper.com/api/1/docs/geocoding/ ",
  "parameters" => [
    {
      "name" => "q",
      "description" => "If you do forward geocoding, then this would be a textual description of the address you are looking for",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "locale",
      "description" => "Display the search results for the specified locale. Currently French (fr), English (en), German (de) and Italian (it) are supported. If the locale wasn&#39;t found the default (en) is used.",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "limit",
      "description" => "Specify the maximum number of returned results",
      "dataType" => "int",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "reverse",
      "description" => "Set to true to do a reverse Geocoding request, see point parameter",
      "dataType" => "boolean",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "point",
      "description" => "The location bias in the format &#39;latitude,longitude&#39; e.g. point&#x3D;45.93272,11.58803",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "provider",
      "description" => "Can be either, default, nominatim, opencagedata",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    {
      "name" => "key",
      "description" => "Get your key at graphhopper.com",
      "dataType" => "string",
      "allowableValues" => "",
      "paramType" => "query",
    },
    ]}) do
  cross_origin
  # the guts live here

  {"message" => "yes, it worked"}.to_json
end

